cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -628,13 +641,21 @@ private List<ApplicationId> decodeAndValidateBatchApplication(NamespaceId namesp
           throw new BadRequestException("Missing 'appId' in the request element.");
         }
         String appId = obj.get("appId").getAsString();
-
-        JsonElement version = obj.get("version");
-        if (version == null) {
-          result.add(validateApplicationId(namespaceId, appId));
+        String version;
+        JsonElement versionObj = obj.get("version");
+        if (versionObj == null || versionObj.equals("-SNAPSHOT")) {
+          // get the latest version of the app instead of -SNAPSHOT
+          try {
+            ApplicationDetail appLatestVersionDetail = applicationLifecycleService
+              .getLatestAppDetail(new ApplicationId(namespaceId.getNamespace(), appId));
+            version = appLatestVersionDetail.getAppVersion();
+          } catch (Exception e) {
```

Comment:
How do you make sure the exception thrown here will always be 404? Should be specific on the exception catch

Reply:
removed the catch block here as per the discussion https://github.com/cdapio/cdap/pull/14627/files#r996174596

Reply:
removed this catch block